docs/blog/posts/2024/10/ci-workflow-for-EESSI.md

```diff
@@ -0,0 +1,48 @@
+---
+authors: [Neves-P]
+date: 2024-10-11
+slug: ci-workflow-for-EESSI 
+---
+
+# CI workflow for EESSI
+
+[EESSI's CI workflows](https://www.eessi.io/docs/using_eessi/eessi_in_ci/) are available on [GitHub Actions](https://github.com/marketplace/actions/eessi) and as a [GitLab CI/CD component](https://gitlab.com/explore/catalog/eessi/gitlab-eessi). Enabling this is as simple as adding EESSI's CI to your workflow of choice, giving you access to the entire EESSI software stack optimized for the relevant CPU architecture(s) in your runner's environment. If you are developing an application on top of the EESSI software stack, for example, this means you don't need to invest heavily in configuring and maintaining a CI setup: EESSI does that for you so you can focus on your code. With the EESSI CI workflows you don't have to worry about figuring out how to optimize build and runtime dependencies as these will be streamed seamlessly to your runner's environment.
+
+<!-- more -->
+
+## Using the CI component in GitLab
+
+To showcase this, let's create a simple `R` package that just outputs a map of the European Union and Norway, and colours the participating countries in the [MultiXscale CoE](https://multixscale.eu).
+
+We'll make a package [`eessirmaps`](https://gitlab.com/neves-p/eessirmaps) that relies on popular R packages [`ggplot2`](https://cran.r-project.org/package=ggplot2), [`sf`](https://cran.r-project.org/package=sf), and [`rnaturalearth`](https://cran.r-project.org/package=rnaturalearth) to render and save this map. Installing GIS tools for R can be somewhat cumbersome, which could become trickier if it has to be done in a CI environment. This is because `sf` requires system packages `libgdal-dev` and `libproj-dev`, which would add yet another step, complicating our CI workflow. Thankfully, EESSI makes a lot of the packages dependencies available to us from the start, as well as a fully functioning version of R, and the necessary package dependencies to boot!. As far as setup goes, this results in a simple CI workflow:
```

Comment:
```suggestion
We'll make a package [`eessirmaps`](https://gitlab.com/neves-p/eessirmaps) that relies on popular R packages [`ggplot2`](https://cran.r-project.org/package=ggplot2), [`sf`](https://cran.r-project.org/package=sf), and [`rnaturalearth`](https://cran.r-project.org/package=rnaturalearth) to render and save this map. Installing GIS tools for R can be somewhat cumbersome, which could become trickier if it has to be done in a CI environment. This is because `sf` requires system packages `libgdal-dev` and `libproj-dev`, which would add yet another step, complicating our CI workflow. Thankfully, EESSI makes a lot of the packages dependencies available to us from the start, as well as a fully functioning version of R, and the necessary package dependencies to boot! As far as setup goes, this results in a simple CI workflow:
```